app/main/views/services.py

```diff
@@ -715,12 +722,35 @@ def set_remove_last_subsection_error_message():
                 ))
 
 
-@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/previous-services', methods=['GET'])
+@main.route('/frameworks/<framework_slug>/submissions/<lot_slug>/previous-services', methods=['GET', 'POST'])
 @login_required
 def list_previous_services(framework_slug, lot_slug):
     framework, lot = get_framework_and_lot_or_404(data_api_client, framework_slug, lot_slug)
     if framework['status'] != 'open':
         abort(404)
+    form = OneServiceLimitCopyServiceForm(lot['name'].lower())
```

Comment:
There's something quite confusing here about having this view be both for listing services and for showing this special form. It feels like really they ought to be separate views, if at all possible - and it looks like the template is really two templates with a massively important `if` to switch between these two cases, as well!

Reply:
My thinking about keeping them together, and perhaps the naming of the view doesn't help, is that there's one view that deals with presenting the supplier with the options for copying their previous services - regardless of the type of lot.

I thought about splitting the template out into partials. Happy to split it into two separate templates though if you think it'd be better?

Reply:
Yep. Maybe only set `form` if `lot.get('oneServiceLimit')`? I know it's not strictly necessary, and constructing that thing won't really impact performance - but it might make it more clear what's going on? You'd have to be careful about how you're passing the errors to the template if you did this.

I'm happy with one template if you're sure you want to stick with one view.

Reply:
Cool, have updated to only load the form if necessary.